lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -797,19 +837,24 @@ def _handle_unschedulable_job(self, job, job_state):
     def _handle_job_failure(self, job, job_state):
         # Figure out why job has failed
         with open(job_state.error_file, "a") as error_file:
+            log.debug("PP Trying with error file in _handle_job_failure")
             if self.__job_failed_due_to_low_memory(job_state):
+                log.debug("PP OOM reached!")
                 error_file.write("Job killed after running out of memory. Try with more memory.\n")
                 job_state.fail_message = "Tool failed due to insufficient memory. Try with more memory."
                 job_state.runner_state = JobState.runner_states.MEMORY_LIMIT_REACHED
             elif self.__job_failed_due_to_walltime_limit(job):
+                log.debug("PP checking for walltime")
                 error_file.write("DeadlineExceeded")
                 job_state.fail_message = "Job was active longer than specified deadline"
                 job_state.runner_state = JobState.runner_states.WALLTIME_REACHED
             else:
-                error_file.write("Exceeded max number of Kubernetes pod retries allowed for job\n")
-                job_state.fail_message = "More pods failed than allowed. See stdout for pods details."
-        job_state.running = False
-        self.mark_as_failed(job_state)
+                log.debug("PP no idea!")
```

Comment:
```suggestion
                log.debug(f"Unknown error detected in job: {job_state.job_id}")
```

Reply:
Since k8s is now handling all errorcodes returned from the job, shouldn't error codes not recognized by k8s be returned to Galaxy?

Reply:
Yes, but I think that that happens when fail_job method invokes the reading of the stderr, stdout and error code files. Or do you have something else in mind?

Reply:
However, the exit code as such is not being captured as part of the failing process (different to stderr and stdout). I remember seeing the code that captured the exit code in the stderr / stdout handling elsewhere. If you know of an easy way of adding that here, we can add, otherwise I can fish that out from the finish job code.

Reply:
![image](https://user-images.githubusercontent.com/368478/225260928-b73523e9-6bb9-41ed-a4e0-1fccf6f87ad1.png)

That particular tool doesn't have stdout in this case, but there should be an exit code there.